### ISSUE TITLE
QueryBuilderEngine.php#294 ErrorException

### ISSUE THREAD

**Issue by dechoiliev** [yajra/laravel-datatables#448]
This is the great package.Thanks.

But i have a problem below:

```php
Invalid argument supplied for foreach()
in QueryBuilderEngine.php line 294
at HandleExceptions->handleError('2', 'Invalid argument supplied for foreach()', '/vendor/yajra/laravel-datatables-oracle/src/Engines/QueryBuilderEngine.php', '294', array('columns' => null)) in QueryBuilderEngine.php line 294
at QueryBuilderEngine->columnSearch() in BaseEngine.php line 579
at BaseEngine->filterRecords() in BaseEngine.php line 543
at BaseEngine->make(true, false) in QueryBuilderEngine.php line 72
at QueryBuilderEngine->make(true) in UsersController.php line 71
at UsersController->anyDatauser()
at call_user_func_array(array(object(UsersController), 'anyDatauser'), array()) in Controller.php line 256
at Controller->callAction('anyDatauser', array()) in ControllerDispatcher.php line 164
at ControllerDispatcher->call(object(UsersController), object(Route), 'anyDatauser') in ControllerDispatcher.php line 112
at ControllerDispatcher->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 139
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 103
at Pipeline->then(object(Closure)) in ControllerDispatcher.php line 114
at ControllerDispatcher->callWithinStack(object(UsersController), object(Route), object(Request), 'anyDatauser') in ControllerDispatcher.php line 69
at ControllerDispatcher->dispatch(object(Route), object(Request), 'Fermer\Http\Controllers\Admin\users\UsersController', 'anyDatauser') in Route.php line 203
at Route->runWithCustomDispatcher(object(Request)) in Route.php line 134
at Route->run(object(Request)) in Router.php line 708
at Router->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 139
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Admin.php line 24
at Admin->handle(object(Request), object(Closure))
at call_user_func_array(array(object(Admin), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 124
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in AuthAdmin.php line 49
at AuthAdmin->handle(object(Request), object(Closure))
at call_user_func_array(array(object(AuthAdmin), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 124
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 103
at Pipeline->then(object(Closure)) in Router.php line 710
at Router->runRouteWithinStack(object(Route), object(Request)) in Router.php line 675
at Router->dispatchToRoute(object(Request)) in Router.php line 635
at Router->dispatch(object(Request)) in Kernel.php line 236
at Kernel->Illuminate\Foundation\Http\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 139
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Debugbar.php line 49
at Debugbar->handle(object(Request), object(Closure))
at call_user_func_array(array(object(Debugbar), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 124
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in VerifyCsrfToken.php line 50
at VerifyCsrfToken->handle(object(Request), object(Closure))
at call_user_func_array(array(object(VerifyCsrfToken), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 124
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in ShareErrorsFromSession.php line 49
at ShareErrorsFromSession->handle(object(Request), object(Closure))
at call_user_func_array(array(object(ShareErrorsFromSession), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 124
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in StartSession.php line 62
at StartSession->handle(object(Request), object(Closure))
at call_user_func_array(array(object(StartSession), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 124
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in AddQueuedCookiesToResponse.php line 37
at AddQueuedCookiesToResponse->handle(object(Request), object(Closure))
at call_user_func_array(array(object(AddQueuedCookiesToResponse), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 124
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in EncryptCookies.php line 59
at EncryptCookies->handle(object(Request), object(Closure))
at call_user_func_array(array(object(EncryptCookies), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 124
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in CheckForMaintenanceMode.php line 44
at CheckForMaintenanceMode->handle(object(Request), object(Closure))
at call_user_func_array(array(object(CheckForMaintenanceMode), 'handle'), array(object(Request), object(Closure))) in Pipeline.php line 124
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 103
at Pipeline->then(object(Closure)) in Kernel.php line 122
at Kernel->sendRequestThroughRouter(object(Request)) in Kernel.php line 87
at Kernel->handle(object(Request)) in index.php line 54
```

**This is my code(previous version it works;maybe v.6.7.3):**
```php
$users = User::select(['id', 'name', 'email', 'subdomain', 'provider', 'ban', 'password', 'created_at']);

        return Datatables::of($users)
            ->addColumn('action', function ($user) {
                $login_as_link = '';
                if(Auth::user()->user_role == 'admin')
                {
                    $login_as_link = '<a href="'.route('login_as', $user->id).'" class="btn btn-success userBan mRight">Login as</a>';
                }
                if($user->ban == 1) {
                    return $login_as_link.'<button type="button" data-user-id="'.$user->id.'" data-user-state="'.$user->ban.'" class="btn btn-success userBan mRight">Activate</button>';
                }else{
                    return $login_as_link.'<button type="button" data-user-id="'.$user->id.'" data-user-state="'.$user->ban.'" class="btn btn-danger userBan mRight">Ban</button>';
                }
            })
            ->editColumn('id', '{{$id}}')
            ->editColumn('provider', '{{ ($provider == "") ? "" : $provider }}')
            ->editColumn('name', '<a href=""> {{ $name }} </a></td>')
            ->removeColumn('password')
            ->make(true);
```

Regards,
Decho

**Comment by yajra** [comment]
@dechoiliev, what version are you using now? Can you please paste your js? Thanks!

**Comment by dechoiliev** [comment]
Package is a v6.9.0 and js is a DataTables 1.10.7

**Comment by yajra** [comment]
I think your js does not submit any `columns` information in the request which is causing the issue. The change was on v6.8.0.

Anyways, can you please try change `QueryBuilderEngine` and update `line 292:` with:
```php
$columns = $this->request->get('columns', []);
````

Let's see if this works and I will make a patch. Thanks!

**Comment by dechoiliev** [comment]
Yes, it works. Can be patch.
Thanks.

**Comment by yajra** [comment]
Patch released on v6.9.1. Thanks for confirming!
